pkg/boot/bzimage/bzimage_decompress.go

```diff
@@ -127,3 +128,18 @@ func unzstd(w io.Writer, r io.Reader) error {
 	}
 	return nil
 }
+
+// unxz reads compressed bytes from the io.Reader and writes the uncompressed bytes to the
+// writer. unxz satisfies the decompressor interface.
+func unxz(w io.Writer, r io.Reader) error {
+	unxzReader, err := xz.NewReader(r, 0)
+	if err != nil {
+		// falling back to shelling out to 'unxz' 
+		unxzReader, nil = execer("unxz")
```

Comment:
Looks like I totally misunderstood how to implement that. Anyone knows where I should implement falling back to shell? I'm guessing in bzimage.go but all those magic bits are inline.

Reply:
I implemented the fixes (thanks for the help @rminnich). We now have fallback to the xz binary. The tests are passing too.